In [11]:
import pandas as pd

df = pd.read_csv("data/data.csv", sep=";", quotechar='"')
print(df.columns.tolist())
df.head()


['ID', 'design_preference', 'age_group', 't_A', 't_B', 'vas_A', 'vas_B']


ID  design_preference  age_group     t_A     t_B  vas_A  vas_B
0   1                  0          1  30.920  22.846  0.665  0.657
1   2                  1          1  14.411  20.367  0.700  0.717
2   3                  1          1  19.116  17.611  0.637  0.688
3   4                  1          0  39.704  28.309  0.595  0.693
4   5                  0          1  32.509  27.516  0.560  0.535

## Aufgabe 1 a

Um zu prüfen, ob der beobachtete Unterschied statistisch signifikant ist, stellt man zunächst eine Nullhypothese auf, die es anschließend zu widerlegen gilt.

## Nullhypothese und Binomialverteilung

Ansetzen kann man hier mit einer Binomialverteilung, bei der unter der Nullhypothese eine perfekte 50/50-Verteilung angenommen wird.

## Signifikanzniveau und p-Wert

Damit wir die Nullhypothese verwerfen können, muss die Wahrscheinlichkeit, dass unsere Datensätze mit der Nullhypothese vereinbar sind (p-Wert), kleiner als 0,05 sein.

---

Jetzt das Ganze im Code:



In [13]:
from scipy.stats import binomtest

n = len(df)
k_1 = (df["design_preference"] == 1).sum()  # Anzahl Design B

test = binomtest(k_1, n, p=0.5, alternative="two-sided")

print("Anzahl Beobachtungen:", n)
print("Anzahl Design B (1):", k_1)
print("Anzahl Design A (0):", n - k_1)
print("p-Wert:", f"{test.pvalue:.3f}")


Anzahl Beobachtungen: 90
Anzahl Design B (1): 54
Anzahl Design A (0): 36
p-Wert: 0.073


Aus dem p-Wert (0.073) und unserem α (0.05) kann man die Nullhypothese nicht verwerfen, woraus sich keine statistisch signifikante Präferenz für eines der beiden UI-Designs in der Grundgesamtheit ableiten lässt.